lib/owl/data.ex

```diff
@@ -296,31 +298,87 @@ defmodule Owl.Data do
 
   defp do_to_ansidata(term, _open_tags), do: term
 
-  defp maybe_wrap_to_tag([], [element]), do: element
-  defp maybe_wrap_to_tag([], data), do: data
+  @doc """
+  Transforms data from `t:IO.ANSI.ansidata/0`, replacing raw escape sequences with tags (see `tag/2`).
 
-  defp maybe_wrap_to_tag(sequences1, [%Owl.Tag{sequences: sequences2, data: data}]) do
-    tag(data, collapse_sequences(sequences1 ++ sequences2))
+  This makes it possible to use data formatted outside of Owl with other Owl modules, like `Owl.Box`.
+
+  The `ansidata` passed to this function must contain escape sequences as separate binaries, not concatenated with other data.
+  For instance, the following will work:
+
+      iex> Owl.Data.from_ansidata(["\e[31m", "hello"])
+      Owl.Data.tag("hello", :red)
+
+  Whereas this will not:
+
+      iex> Owl.Data.from_ansidata("\e[31mhello")
+      "\e[31mhello"
+
+  ## Examples
+
+      iex> [[[[[[[] | "\e[46m"] | "\e[31m"], "hello"] | "\e[39m"] | "\e[49m"] | "\e[0m"]
+      ...> |> Owl.Data.from_ansidata()
+      Owl.Data.tag("hello", [:cyan_background, :red])
+
+      iex> [:red, "hello"] |> IO.ANSI.format() |> Owl.Data.from_ansidata()
+      Owl.Data.tag("hello", :red)
+
+  """
+  @spec from_ansidata(IO.ANSI.ansidata()) :: t()
+  def from_ansidata(ansidata) do
+    {data, _open_tags} = do_from_ansidata(ansidata, %{})
+    data
   end
 
-  defp maybe_wrap_to_tag(sequences, data) do
-    tag(data, collapse_sequences(sequences))
+  defp do_from_ansidata(binary, open_tags) when is_binary(binary) do
+    case Sequence.ansi_to_type(binary) do
+      :reset ->
+        {[], %{}}
+
+      nil ->
+        {tag_all(binary, open_tags), open_tags}
+
+      type ->
+        {[], update_open_tags(open_tags, type, Sequence.ansi_to_name(binary))}
+    end
   end
 
-  defp reverse_and_tag(sequences, [%Owl.Tag{sequences: last_sequences} | _] = data) do
-    maybe_wrap_to_tag(sequences -- last_sequences, Enum.reverse(data))
+  defp do_from_ansidata(integer, open_tags) when is_integer(integer) do
```

Comment:
Looking at coverage and this line is red. This is for char list, right?

Reply:
Yeah, it’s just to handle individual character literals in ansidata. `[“hello”, \?]`

Should’ve added a test, sorry!